slither/tools/read_storage/read_storage.py

```diff
@@ -258,6 +330,60 @@ def get_all_storage_variables(self, func: Callable = None) -> None:
                     ],
                 )
             )
+            self._constant_storage_slots.extend(
```

Comment:
Can you loop over `contract.state_variables_ordered`  once instead of twice here and append the relevant variables to `_target_variables ` and `_constant_storage_slots `?

Reply:
Do you mean like this?
```python
    def get_all_storage_variables(self, func: Callable = None) -> None:
        """Fetches all storage variables from a list of contracts.
        kwargs:
            func (Callable, optional): A criteria to filter functions e.g. name.
        """
        for contract in self.contracts:
            for var in contract.state_variables_ordered:
                if func is None or (func is not None and func(var)):
                    if not var.is_constant and not var.is_immutable:
                        self._target_variables.append((contract, var))
                    elif var.is_constant and str(var.type) == "bytes32":
                        self._constant_storage_slots.append((contract, var))
            hardcoded_slot = self.find_hardcoded_slot_in_fallback(contract)
            if hardcoded_slot is not None:
                self._constant_storage_slots.append((contract, hardcoded_slot))
```

Reply:
yeah. also `str(var.type) == "bytes32"` can be `var.type == ElementaryType("bytes32")`

Reply:
Alright, I've made these changes